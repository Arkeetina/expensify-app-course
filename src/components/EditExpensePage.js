import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {startEditExpense, startRemoveExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onClick = (id) => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }

  render () {
    return (
      <div>
        <button onClick={this.onClick}>Remove</button>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  startRemoveExpense: (id) => dispatch(startRemoveExpense(id)),
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense))
});

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id )
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
