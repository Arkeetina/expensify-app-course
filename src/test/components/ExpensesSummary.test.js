import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
 

test('should render ExpensesSummary component correctly with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary component correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={3333}/>);
  expect(wrapper).toMatchSnapshot();
});

