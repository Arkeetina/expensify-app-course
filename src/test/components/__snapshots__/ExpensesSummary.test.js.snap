// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary component correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     expenses totalling 
    $33.33
  </h1>
</div>
`;

exports[`should render ExpensesSummary component correctly with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     expenses totalling 
    $2.35
  </h1>
</div>
`;
